File: backend/src/main/java/com/decmoe47/todo/service/TodoService.java
package com.decmoe47.todo.service;

import com.decmoe47.todo.model.dto.*;
import com.decmoe47.todo.model.vo.TodoVO;

import java.util.List;

public interface TodoService {

    List<TodoVO> getTodos(String listId);

    TodoVO addTodo(TodoAddDTO todoAddDTO);

    void deleteTodos(List<TodoDeleteDTO> todoDeleteDTOs);

    List<TodoVO> updateTodos(List<TodoUpdateDTO> todoUpdateDTOS);

    TodoVO toggleTodo(TodoToggleDTO todoToggleDTO);

    List<TodoVO> moveTodos(List<TodoMoveDTO> todoMoveDTOs);
}

File: backend/src/main/kotlin/com/decmoe47/todo/service/TodoService.kt
package com.decmoe47.todo.service

import com.decmoe47.todo.model.request.*
import com.decmoe47.todo.model.response.TodoResponse

interface TodoService {
    fun getTodos(listId: Long): List<TodoResponse>

    fun addTodo(request: TodoAddRequest): TodoResponse

    fun deleteTodo(request: TodoDeleteRequest)

    fun updateTodo(request: TodoUpdateRequest): TodoResponse

    fun toggleTodo(request: TodoToggleRequest): TodoResponse

    fun moveTodo(request: TodoMoveRequest): TodoResponse
}
